### PR TITLE
bug( TDP-4572  TDQ-14421 ) use 'ResolverStyle.STRICT' to parse a Date.

### PR DESCRIPTION
**Pull request by qiongli**
https://jira.talendforge.org/browse/TDP-4572

### CODE REVIEW

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/865/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/865/)

File: dataprep-backend-common/src/main/java/org/talend/dataprep/transformation/actions/date/DateParser.java [hunk 1]
```diff
@@ -146,8 +147,14 @@ public LocalDateTime parseDateFromPatterns(String value, List<DatePattern> patte
         }
 
         for (DatePattern pattern : patterns) {
-            final DateTimeFormatter formatter = new DateTimeFormatterBuilder().parseCaseInsensitive()
-                    .append(pattern.getFormatter()).toFormatter(Locale.ENGLISH);
+            // TDQ-14421 use ResolverStyle.STRICT to parse a date. such as "2017-02-29" should be
+            // invalid.STRICT model for pattern without G,should replace 'y' with 'u'.
+            String patternStr = pattern.getPattern();
+            if (!StringUtils.endsWith(patternStr, "G")) {
```
Comment (hmdebenque) [inline dataprep-backend-common/src/main/java/org/talend/dataprep/transformation/actions/date/DateParser.java]: This is creating a divergence between our date patterns and the java standard.
If our patterns are not correct, we should change them.
If theses kind of patterns are not user-friendly enough, we should convert them at frontend level.

Comment (qiongli) [reply dataprep-backend-common/src/main/java/org/talend/dataprep/transformation/actions/date/DateParser.java]: Yes,I have thought about this.  if we decide to change the pattern in lower level, we might need to update all date patterns in DQ lib. but this big change will impact all related Date Functions.
QA must test all realted cases. In the other hand, The "u" symbol represent "year" is a new property on Java8, on DP UI,customer can input themself pattern,they are used to "yyyy". So I think the repalce 'y ' with 'u' is necessary.
We can use the replace solution to fix the customer issue. Then we create a new workitem to update our patterns. Do you agree?

Comment (scorreia) [reply dataprep-backend-common/src/main/java/org/talend/dataprep/transformation/actions/date/DateParser.java]: Replacing "y" by "u" is done at two places: in DQ lib and in the callers of the DQ lib. 
Clearly, this cannot be kept like this. 
All applications that would like to use the DQ lib don't know that they must replace "y" by "u". 
That should never be their responsibility to do so.

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/912/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/912/)

**Pull request closed without merging**